ichnaea/data/tests/test_validation.py

```diff
@@ -256,7 +231,161 @@ def test_asu_signal_field_mix_up(self):
+            (measure, cell) = self.make_cell_submission(**entry)
+            self.check_normalized_cell(measure, cell, None)
+
+    def test_records_fail_the_mnc_check(self):
+        entries = [
+            {'mcc': FRANCE_MCC, 'mnc': -1, 'lac': 3, 'cid': 4},
+            {'mcc': FRANCE_MCC, 'mnc': -2, 'lac': 3, 'cid': 4},
+            {'mcc': FRANCE_MCC, 'mnc': 33000, 'lac': 3, 'cid': 4},
+
+        ]
+
+        for entry in entries:
+            if 'radio' not in entry:
+                entry['radio'] = 'cdma'
+            (measure, cell) = self.make_cell_submission(**entry)
+            self.check_normalized_cell(measure, cell, None)
+
+    def test_records_fail_the_lac_check(self):
+        entries = [
+            {'mcc': FRANCE_MCC, 'mnc': 2, 'lac': -1, 'cid': 4},
+            {'mcc': FRANCE_MCC, 'mnc': 2, 'lac': -2, 'cid': 4},
+            {'mcc': FRANCE_MCC, 'mnc': 2, 'lac': 65536, 'cid': 4},
+
+        ]
+
+        for entry in entries:
+            if 'radio' not in entry:
+                entry['radio'] = 'cdma'
+            (measure, cell) = self.make_cell_submission(**entry)
+            self.check_normalized_cell(measure, cell, None)
+
+    def test_records_fail_the_cid_check(self):
+        entries = [
+            {'mcc': FRANCE_MCC, 'mnc': 2, 'lac': 3, 'cid': -1},
+            {'mcc': FRANCE_MCC, 'mnc': 2, 'lac': 3, 'cid': -2},
+            {'mcc': FRANCE_MCC, 'mnc': 2, 'lac': 3, 'cid': 2 ** 28},
+
+        ]
+
+        for entry in entries:
+            if 'radio' not in entry:
+                entry['radio'] = 'cdma'
+            (measure, cell) = self.make_cell_submission(**entry)
+            self.check_normalized_cell(measure, cell, None)
+
+    def test_records_fail_the_lac_or_cid_and_psc_check(self):
+        entries = [
+            {'mcc': FRANCE_MCC, 'mnc': 2, 'lac': -1, 'cid': -1},
+            {'mcc': FRANCE_MCC, 'mnc': 2, 'lac': 3, 'cid': -1},
+            {'mcc': FRANCE_MCC, 'mnc': 2, 'lac': -1, 'cid': 4},
+
+        ]
+
+        for entry in entries:
+            if 'radio' not in entry:
+                entry['radio'] = 'cdma'
+            (measure, cell) = self.make_cell_submission(**entry)
+            self.check_normalized_cell(measure, cell, None)
+
+    def test_unknown_lac_cid_is_65535_and_missing_psc(self):
+        entries = [
+            {'mcc': FRANCE_MCC, 'mnc': 2, 'lac': 0, 'cid': 65535, 'psc': -1},
+
+        ]
+
+        for entry in entries:
+            if 'radio' not in entry:
+                entry['radio'] = 'cdma'
+            (measure, cell) = self.make_cell_submission(**entry)
+            self.check_normalized_cell(measure, cell, None)
+
+    def test_fails_because_it_has_an_MNC_above_1000_for_a_GSM_network(self):
+        entries = [
+            {'mcc': FRANCE_MCC, 'mnc': 1001, 'lac': 3, 'cid': 4,
+             'radio': 'gsm'},
+        ]
+
+        for entry in entries:
+            if 'radio' not in entry:
+                entry['radio'] = 'cdma'
```

Comment:
here we can just take away these two lines, as we specify radio explicitly in the above entry